RadClass/DiffSpectra.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+
+
+class DiffSpectra:
+    '''
+    Estimates a minimum background spectrum for each
+    timestamp/spectrum and saves a difference spectrum
+    between the two.
+
+    Attributes:
+    stride: The number of timestamps prior to the current
+        spectrum for finding the minimum background
+    '''
+
+    def __init__(self, stride=10):
+        self.stride = stride
+
+    def run(self, data):
+        # index all windows of length self.diff_stride into a tmp array
+        windows = [data[i-self.stride:i] for i in range(self.stride, data.shape[0])]
```

Comment:
This seems very memory intensive without a lot of benefit other than this compact notation.  Can we discuss at a software meeting?

Reply:
Good idea and agreed. Ideally I would think these lists could be combined into one line, rather than building one list to build another etc.  Maybe in a way that could better utilize NumPy/arrays rather than lists.

At the very least, it's important to avoid building a list of windows. That is ballooning the memory requirements since identical timestamps will be copied to `diff_stride` windows. Instead, the min/argmin should be calculated outright.